Services/WishListServices.cs

```diff
@@ -0,0 +1,312 @@
+                .Where(epr => epr.WishListId == wishlist.ContentItem.Id)
+                .List();
+        }
+
+        public bool ItemIsInWishlist(WishListListPart wishlist, ShoppingCartItem item) {
+            return GetElements(wishlist).Any(wel =>
+                ShoppingCartItem.ItemsAreEqual(wel.Item, item));
+        }
+
+        public void RemoveElementFromWishlist(IUser user, WishListListPart wishlist, int elementId) {
+            if (user == null) {
+                throw new ArgumentNullException("user");
+            }
+            if (wishlist.ContentItem.As<CommonPart>().Owner != user) {
+                throw new ArgumentException(T("Users may only edit wishlists they own.").Text);
+            }
+
+            var elementPart = _contentManager.Get<WishListElementPart>(elementId);
+            
+            RemoveElementFromWishlist(wishlist, elementPart);
+        }
+        private void RemoveElementFromWishlist(WishListListPart wishlist, WishListElementPart elementPart) {
+            //process extensions
+            foreach (var ext in _wishListExtensionProviders) {
+                ext.WishListElementCleanup(wishlist, elementPart);
+            }
+
+            var elementId = elementPart.ContentItem.Id;
+            if (wishlist.Ids.Contains(elementId)) {
+                var elementIds = wishlist.Ids.ToList().Where(id => id != elementId);
+                wishlist.Ids = elementIds.ToArray();
+            }
+            _contentManager.Destroy(elementPart.ContentItem); //hard delete
+        }
+
+        public void DeleteWishlist(IUser user, WishListListPart wishlist) {
+            if (user == null) {
+                throw new ArgumentNullException("user");
+            }
+            if (wishlist.ContentItem.As<CommonPart>().Owner != user) {
+                throw new ArgumentException(T("Users may only edit wishlists they own.").Text);
+            }
+
+            //remove all elements
+            foreach (var element in GetElements(wishlist)) {
+                RemoveElementFromWishlist(wishlist, element);
+            }
+            //destroy wishlist
+            //process extensions
+            foreach (var ext in _wishListExtensionProviders) {
+                ext.WishListCleanup(user, wishlist);
+            }
+            if (wishlist.IsDefault) {//user must still have a default
+                var otherLists = GetWishLists(user).Where(wl => wl.ContentItem.Id != wishlist.ContentItem.Id);
+                if (otherLists.Any()) {
+                    otherLists.First().IsDefault = true;
+                } else {
+                    GetDefaultWishList(user); //create a new default wishlist
+                }
+                wishlist.IsDefault = false;
+            }
+            _contentManager.Destroy(wishlist.ContentItem);
+        }
+
+        public dynamic CreateShape(IUser user, ProductPart product = null) {
```

Comment:
Oh well, I said static, but it doesn't have to be, if it would be too many parameters to pass to the methods. I would still separate into something like a `WishListUIService`.